src/authenticationservices.cs

```diff
@@ -930,4 +951,342 @@ interface ASAccountAuthenticationModificationViewController {
+		NSString UserVerificationPreference { get; set; }
+
+		[Abstract]
+		[Export ("attestationPreference")]
+		NSString AttestationPreference { get; set; }
+	}
+
+	[NoWatch, Mac (12,0), iOS (15,0), MacCatalyst (15,0), NoTV]
+	[Protocol]
+	interface ASPublicKeyCredential : ASAuthorizationCredential
+	{
+		[Abstract]
+		[Export ("rawClientDataJSON", ArgumentSemantic.Copy)]
+		NSData RawClientDataJson { get; }
+
+		[Abstract]
+		[Export ("credentialID", ArgumentSemantic.Copy)]
+		NSData CredentialId { get; }
+	}
+
+	[NoWatch, Mac (12,0), iOS (15,0), MacCatalyst (15,0), NoTV]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface ASAuthorizationPlatformPublicKeyCredentialDescriptor : ASAuthorizationPublicKeyCredentialDescriptor
+	{
+		[Export ("initWithCredentialID:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSData credentialId);
+	}
+
+	[NoWatch, Mac (12,0), iOS (15,0), MacCatalyst (15,0), NoTV]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface ASAuthorizationPlatformPublicKeyCredentialProvider : ASAuthorizationProvider
+	{
+		[Export ("initWithRelyingPartyIdentifier:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (string relyingPartyIdentifier);
+
+		[Export ("createCredentialRegistrationRequestWithChallenge:name:userID:")]
+		ASAuthorizationPlatformPublicKeyCredentialRegistrationRequest CreateCredentialRegistrationRequestWithChallenge (NSData challenge, string name, NSData userId);
+
+		[Export ("createCredentialAssertionRequestWithChallenge:")]
+		ASAuthorizationPlatformPublicKeyCredentialAssertionRequest CreateCredentialAssertionRequestWithChallenge (NSData challenge);
+
+		[Export ("relyingPartyIdentifier")]
+		string RelyingPartyIdentifier { get; }
+	}
+
+	[NoWatch, NoTV, Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface ASAuthorizationProviderExtensionAuthorizationResult
+	{
+		[Export ("initWithHTTPAuthorizationHeaders:")]
+		IntPtr Constructor (NSDictionary<NSString, NSString> httpAuthorizationHeaders);
+
+		[Export ("initWithHTTPResponse:httpBody:")]
+		IntPtr Constructor (NSHttpUrlResponse httpResponse, [NullAllowed] NSData httpBody);
+
+		[NullAllowed, Export ("httpAuthorizationHeaders", ArgumentSemantic.Assign)]
+		NSDictionary<NSString, NSString> HttpAuthorizationHeaders { get; set; }
+
+		[NullAllowed, Export ("httpResponse", ArgumentSemantic.Copy)]
+		NSHttpUrlResponse HttpResponse { get; set; }
+
+		[NullAllowed, Export ("httpBody", ArgumentSemantic.Assign)]
+		NSData HttpBody { get; set; }
+
+		[Export ("privateKeys", ArgumentSemantic.Assign)]
+		NSObject[] PrivateKeys { get; set; }
+	}
+
+	[NoWatch, NoTV, Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface ASAuthorizationPublicKeyCredentialParameters : NSSecureCoding, NSCopying
+	{
+		[Export ("initWithAlgorithm:")]
+		IntPtr Constructor (nint algorithm);
+
+		[Export ("algorithm")]
+		nint Algorithm { get; }
```

Comment:
```
+@property (nonatomic, readonly) ASCOSEAlgorithmIdentifier algorithm;
```

why not use the type ?

Reply:
I cannot find this type in our bindings using grep

Reply:
it might be new... not sure why xtro did not pick it up
you need to track the header files to see/learn why :)

Reply:
Hmm okay so the headers didn't really reveal anything to me: ```@abstract A COSE algorithm indentifier.```
 but looking at the web docs and looking up what ASCOSEAlgorithmIdentifier reveals that it is a type alias and has this:
```typedef NSInteger ASCOSEAlgorithmIdentifier;```
But I am not sure if this is any better?
https://developer.apple.com/documentation/authenticationservices/ascosealgorithmidentifier?language=objc

Reply:
I'll keep looking at this after the meeting though

Reply:
It's not how most enums are defined - and it seems sharpie does not process them when defined this way.

```
API_AVAILABLE(macos(12.0), ios(15.0)) API_UNAVAILABLE(watchos)
typedef NSInteger ASCOSEAlgorithmIdentifier NS_TYPED_EXTENSIBLE_ENUM;

/*! @abstract The COSE algorithm identifier representing ECDSA with SHA-256.
 */
API_AVAILABLE(macos(12.0), ios(15.0)) API_UNAVAILABLE(watchos)
static ASCOSEAlgorithmIdentifier const ASCOSEAlgorithmIdentifierES256 = -7;
```

/Applications/Xcode_13.0.0-beta3.app//Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS.sdk/System/Library/Frameworks/AuthenticationServices.framework/Headers/ASCOSEConstants.h 

Also check for `ASCOSEEllipticCurveIdentifier` since it might suffer from the same `nine` issue

Reply:
AH thanks for finding this! I will grep the header files next time!

Reply:
Interestingly, ASCOSEEllipticCurveIdentifier is only declared but not used

Reply:
some new (b4?) API might eventually use it
but no need to bind it until we have a need for it :)